tests/regression/test_star_pc.py

```diff
@@ -20,6 +21,12 @@ def backend(request):
     return request.param
 
 
+def filter_warnings(caller):
+    with warnings.catch_warnings():
+        warnings.filterwarnings("error", "Creating new TransferManager", RuntimeWarning)
+        caller()
```

Comment:
Can you leave a comment explaining exactly what this code is doing. It looks like it is going to have unintended consequences.

From my reading of the Python docs this function catches and suppresses all warnings, except for those that match "Creating new TransferManager" in the message component, in which case an exception is raised instead. However, I might have misunderstood the behaviour.

Reply:
This will still raise other warnings without supressing them.

```
import warnings

def filter_warnings(caller):
    with warnings.catch_warnings():
        warnings.filterwarnings("error", "Creating new TransferManager", RuntimeWarning)
        caller()

def fatal_warning():
    warnings.warn("Creating new TransferManager to transfer data to coarse grids", RuntimeWarning)

def non_fatal_warning():
    warnings.warn("Should be raised", RuntimeWarning)

filter_warnings(non_fatal_warning)
print("That was just a warning")

filter_warnings(fatal_warning)
print("We should not be here")
```
Gives the expected output

```
/home/brubeckmarti/Downloads/warnings.py:7: RuntimeWarning: Should be raised
  warnings.warn("Should be raised", RuntimeWarning)
That was just a warning
Traceback (most recent call last):
  File "/home/brubeckmarti/Downloads/warnings.py", line 19, in <module>
    filter_warnings(fatal_warning)
  File "/home/brubeckmarti/Downloads/warnings.py", line 13, in filter_warnings
    caller()
  File "/home/brubeckmarti/Downloads/warnings.py", line 4, in fatal_warning
    warnings.warn("Creating new TransferManager to transfer data to coarse grids", RuntimeWarning)
RuntimeWarning: Creating new TransferManager to transfer data to coarse grids

```